### PR TITLE
Ingest unlimited number of documents

### PR DESCRIPTION
**Pull request by andreakiro**
Addressing thread #69 where we noted codebase would ingest only one document from `source_directory`

```python
# Load document and split in chunks
for root, dirs, files in os.walk("source_documents"):
    for file in files:
        if file.endswith(".txt"):
            loader = TextLoader(os.path.join(root, file), encoding="utf8")
        elif file.endswith(".pdf"):
            loader = PDFMinerLoader(os.path.join(root, file))
        elif file.endswith(".csv"):
            loader = CSVLoader(os.path.join(root, file))
documents = loader.load() # loads only the last file content! 
```

@imartinez it would be interesting to think if it is possible to optimize the creation of the `LLAMA` embedding and the `Chroma` db overall. It was already taking quite some time with a single document, and quickly becomes prohibitively slow with more and more docs...

### CODE REVIEW

**Comment by andreakiro** [comment]
@imartinez I would keep the optimization stuff for another PR and already merge this one though.

**Review by omerta222 — APPROVED** [review]

**Review by djsavvy — APPROVED** [review]

File: ingest.py [hunk 1]
```diff
@@ -1,35 +1,62 @@
...
-            elif file.endswith(".pdf"):
-                loader = PDFMinerLoader(os.path.join(root, file))
-            elif file.endswith(".csv"):
-                loader = CSVLoader(os.path.join(root, file))
-    documents = loader.load()
+
+    # Load documents and split in chunks
+    print(f"Loading documents from {source_directory}")
+    documents = load_documents(source_directory)
     text_splitter = RecursiveCharacterTextSplitter(chunk_size=500, chunk_overlap=50)
```
Comment (imartinez) [inline ingest.py:47]: I feel we should be extracting the chunk size and chunk overlap to environmental variables, given both are key iteration parameters to test. Let's leave it for another PR though

File: ingest.py [hunk 2]
```diff
@@ -1,35 +1,62 @@
...
+    csv_files = glob.glob(os.path.join(source_dir, "**/*.csv"), recursive=True)
+    all_files = txt_files + pdf_files + csv_files
+    return [load_single_document(file_path) for file_path in all_files]
+
+
 def main():
-    llama_embeddings_model = os.environ.get('LLAMA_EMBEDDINGS_MODEL')
+    # Load environment variables
     persist_directory = os.environ.get('PERSIST_DIRECTORY')
+    source_directory = os.environ.get('SOURCE_DIRECTORY', 'source_documents')
```
Comment (imartinez) [inline ingest.py:40]: We should probably be creating SOURCE_DIRECTORY variable in the .env file if we are referencing it here, even if it has a default value.

**Review by imartinez — APPROVED** [review]
Great contribution!
